db/migrations/postgres/000026_create_tokenaccount_table.up.sql

```diff
@@ -0,0 +1,15 @@
+BEGIN;
+CREATE TABLE tokenaccount (
+  seq            SERIAL          PRIMARY KEY,
+  namespace      VARCHAR(64)     NOT NULL,
+  pool_id        UUID            NOT NULL,
```

Comment:
I'm storing a `namespace` and a `pool_id` here. That's slightly redundant (since pool IDs will be globally unique). However, it seems that almost every items stores its namespace, so I followed that convention.

There's also an alternate possibility of storing `namespace` and `name`, which are also guaranteed to be unique. Then the `pool_id` would be literally useless and could likely be dropped.

Reply:
The pool ID is generated by FireFly when the pool is created. My point was that I'm not being consistent about how to identify a pool. You can uniquely identify a pool by pool ID, or by namespace + name - so it basically has two primary keys.

The API routes for pools are specified in terms of the namespace + name. In the database here, I'm relating a token account to a given pool using namespace + pool ID. Just felt a little inconsistent once I finished coding it up...

Reply:
Most objects _intentionally_ have these two keys. The distinction (that's lived throughout the KAT days) is:
- The UUID is generated by the submitter of the object into the network. It lets you distinguish between two objects that were independently submitted, and to see if you "won the race" to establish the object.
- The name+namespace combination are the unique lookup key that an application can use in any environment (dev/test/preprod etc.) to refer to the same object in the network.

Reply:
e.g. the UUID is about giving the chain of custody linkage (including contributing to hash uniquness in various places as part of that). But you're right to an application it's kind of redundant at the point of use, and I think that's ok.

Reply:
Discussed with @peterbroadhurst offline, and it seems that namespace and hash are not needed. Probably simplest to start by tracking a compound key of pool ID (or protocol ID?), token index, and identity.